File: FakturusTrack.Android/app/src/main/java/com/fakturus/track/services/subscription/Tier.kt
package com.fakturus.track.services.subscription

enum class Tier(val level: Int) : Comparable<Tier> {
    FREE(0),
    STARTER(1),
    PRO(2);

    val displayName: String
        get() = when (this) {
            FREE -> "Free"
            STARTER -> "Starter"
            PRO -> "Pro"
        }

    companion object {
        fun fromProductId(productId: String): Tier? = when (productId) {
            "starter_monthly" -> STARTER
            "pro_monthly" -> PRO
            else -> null
        }
    }
}

File: FakturusTrack.iOS/FakturusTrack/Services/Subscription/Tier.swift
import Foundation

enum Tier: Int, Comparable, Codable, Sendable {
    case free = 0
    case starter = 1
    case pro = 2

    static func < (lhs: Tier, rhs: Tier) -> Bool {
        lhs.rawValue < rhs.rawValue
    }

    /// Product-ID Mapping (StoreKit 2)
    init?(productID: String) {
        switch productID {
        case "starter_monthly": self = .starter
        case "pro_monthly": self = .pro
        default: return nil
        }
    }

    var displayName: String {
        switch self {
        case .free: "Free"
        case .starter: "Starter"
        case .pro: "Pro"
        }
    }
}
